lib/matplotlib/axes/_base.py

```diff
@@ -2199,53 +2199,22 @@ def autoscale_view(self, tight=None, scalex=True, scaley=True):
         case, use :meth:`matplotlib.axes.Axes.relim` prior to calling
         autoscale_view.
         """
-        if tight is None:
-            _tight = self._tight
-        else:
-            _tight = self._tight = bool(tight)
+        if tight is not None:
+            self._tight = bool(tight)
 
         if self._xmargin or self._ymargin:
-            margins = {
-                'top': True,
-                'bottom': True,
-                'left': True,
-                'right': True
-            }
-            for artist_set in [self.collections, self.patches, self.lines,
-                               self.artists, self.images]:
-                for artist in artist_set:
-                    artist_margins = artist.margins
-                    for key in ['left', 'right', 'top', 'bottom']:
-                        margins[key] &= artist_margins.get(key, True)
-
-            if self._xmargin:
-                for axes in self._shared_x_axes.get_siblings(self):
-                    for artist_set in [axes.collections, axes.patches,
-                                       axes.lines, axes.artists, axes.images]:
-                        for artist in artist_set:
-                            artist_margins = artist.margins
-                            for key in ['left', 'right']:
-                                margins[key] &= artist_margins.get(key, True)
-
-            if self._ymargin:
-                for axes in self._shared_y_axes.get_siblings(self):
-                    for artist_set in [axes.collections, axes.patches,
-                                       axes.lines, axes.artists, axes.images]:
-                        for artist in artist_set:
-                            artist_margins = artist.margins
-                            for key in ['top', 'bottom']:
-                                margins[key] &= artist_margins.get(key, True)
-        else:
-            margins = {
-                'top': False,
-                'bottom': False,
-                'left': False,
-                'right': False
-            }
+            stickies = [artist.sticky_edges for artist in self.get_children()]
+            x_stickies = sum([sticky.x for sticky in stickies], [])
+            y_stickies = sum([sticky.y for sticky in stickies], [])
+            if self.get_xscale().lower() == 'log':
```

Comment:
@nelleV, you think that if a histogram plot is switched to a log scale it should fail rather than yielding a plot?  In other words, simply bumping the "L" key when a histogram plot is active should raise an exception?